test/scss/_assert.scss

```diff
@@ -217,4 +215,139 @@
       }
     }
   }
+
+  @include test-module('Assert (contains)') {
+    @include test('Passes if the property in "contains" exists in the "output" and has the same value') {
+      @include assert {
+        @include output {
+          height: 10px;
+          width: 20px;
+        }
+
+        @include contains {
+          height: 10px;
+        }
+      }
+    }
+
+    @include test('Passes if the property and its value in class in "contains" exists in the same class in "output"') {
+      @include assert {
+        @include output {
+          height: 20px;
+          .class {
+            height: 10px;
+          }
+          .other-class {
+            height: 10px;
+          }
+        }
+
+        @include contains {
+          .class {
+            height: 10px;
+          }
+        }
+      }
+    }
+
+    @include test('Can be used with nested classes') {
+      @include assert {
+        @include output {
+          height: 20px;
+          .class {
+            height: 10px;
+            .other-class {
+              height: 10px;
+            }
+          }
+        }
+
+        @include contains {
+          .class {
+            .other-class {
+              height: 10px;
+            }
+          }
+        }
+      }
+    }
+
+    @include test('Can be used with nested @media queries') {
+      @include assert {
+        @include output {
+          .class{
+            height:20px;
+          }
+          @media (min-width: 30em) {
+            @media (min-width: 40em) {
+              .selector {
+                height: 10px;
+              }
+              .selector2 {
+                height: 10px;
+              }
+            }
+            .selector3 {
+              height: 10px;
+            }
+          }
+        }
+
+        @include contains {
+          @media (min-width: 30em) {
+            @media (min-width: 40em) {
+              .selector2 {
+                height: 10px;
+              }
+            }
+          }
+        }
+      }
+    }
+
+    @include test('Can be used with @at-root') {
+      @include assert {
+        @include output {
+          @at-root {
+            .selector {
+              height: 10px;
+            }
+          }
+        }
+
+        @include contains {
+          @at-root {
+            .selector {
+              height: 10px;
+            }
+          }
+        }
+      }
+    }
+
+    @include test('Can be used for all selector types') {
+      @include assert {
+        @include output {
+          .selector {
```

Comment:
I imagine this also works with attribute selectors, and pseudo-selectors? May be worth adding tests for those? What about *, >, ~, etc?

Reply:
It works for these selectors :-)